Tests/MockInstantiationRewriterTests.cs

```diff
@@ -0,0 +1,43 @@
+﻿//  Copyright (c) rubicon IT GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using NUnit.Framework;
+using RhinoMocksToMoqRewriter.Core;
+
+namespace RhinoMocksToMoqRewriter.Tests
+{
+  [TestFixture]
+  public class MockInstantiationRewriterTests
+  {
+    [Test]
+    [TestCase ("MockRepository.GenerateMock<string>();", "new Mock<string>()")]
+    [TestCase ("MockRepository.GenerateMock<string, IDisposable>();", "new Mock<string>()")]
+    [TestCase ("MockRepository.GenerateMock<string, IDisposable, IConvertible>();", "new Mock<string>()")]
+    [TestCase ("MockRepository.GeneratePartialMock<string>(42);", "new Mock<string>(42)")]
+    [TestCase ("MockRepository.GeneratePartialMock<string>();", "new Mock<string>()")]
+    [TestCase ("MockRepository.GenerateStrictMock<string>();", "new Mock<string>(MockBehaviour.Strict)")]
+    [TestCase ("MockRepository.GenerateStrictMock<string>(42);", "new Mock<string>(MockBehaviour.Strict, 42)")]
+    [TestCase ("MockRepository.GenerateStrictMock<string, IDisposable>();", "new Mock<string>(MockBehaviour.Strict)")]
+    [TestCase ("MockRepository.GenerateStub<string>();", "new Mock<string>()")]
+    [TestCase ("GetSymbolInfo()", "GetSymbolInfo()")]
+    public void MockInstantiationTests (string source, string expected)
```

Comment:
This method should rather be called `Rewrite_ExpressionStatement`, additional tests for the following scenarios should be added:

```csharp
// field
class A
{
  private string _mock = MockRepository.GenerateMock<string>();
}
```

```csharp
// local declaration
class A
{
  public void Do()
  {
    var mock = MockRepository.GenerateMock<string>();
  }
}
```

+ both of the above with different formattings, e.g. a space before the argumentlist and multiline arguments.